src/Asset.js

```diff
@@ -77,9 +77,16 @@ class Asset {
       './' + path.relative(path.dirname(this.name), resolved),
       Object.assign({dynamic: true}, opts)
     );
-    return this.options.parser
-      .getAsset(resolved, this.package, this.options)
-      .generateBundleName();
+    let asset = this.options.parser.getAsset(
+      resolved,
+      this.package,
+      this.options
+    );
+
+    if (this.isMainFile(resolved)) {
```

Comment:
Can you do this check inside `generateBundleName` instead of here? Maybe get rid of the boolean parameter all together and replace with this check?